middleware/src/handlers/handlers.go

```diff
@@ -52,6 +56,21 @@ func (handlers *Handlers) rootHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (handlers *Handlers) getEnvHandler(w http.ResponseWriter, r *http.Request) {
+	data, err := json.Marshal(handlers.middleware.GetSystemEnv())
+	if err != nil {
+		log.Println(err.Error() + " Failed to serialize GetSystemEnv data to json in getEnvHandler")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Perhaps just giving a generic "Oops. Something went wrong" back to the `ResponseWriter` on this line would be best, with the full error details in the logs on L62? That way, as defensive coding, if we had some sensitive information contained in the `err` we'd avoid sending it back over the wire to the caller.

(And elsewhere.)

Reply:
Lets implement proper error handling in the http server in a separate pull request.

Reply:
Okay, fair. How about a `TODO` mentioning that this can be improved for the current PR, then?

Reply:
Ah, I just saw you did change the message returned to the client to `Something went wrong, I cannot read these hieroglyphs`. This is fine as-is, then.